distribution/lib/Standard/Table/0.0.0-dev/src/Data/Aggregate_Column.enso

```diff
@@ -0,0 +1,158 @@
+                First c _ -> "First " + col_name c
+                Last c _ -> "Last " + col_name c
+
+    ## PRIVATE
+       Given a column reference resolve to the underlying column
+    resolve_column : Table->(Column|Text|Integer)->Column
+    resolve_column table column = case column of
+        Column _ -> table.at (column.name)
+        Text -> table.at column
+        Integer -> table.columns.at column
+
+    initial_value : Any
+    initial_value = case this of
+        Count_Distinct _ _ -> Map.empty
+        Median _ _ -> Map.empty
+        Average _ _ -> [0, 0]
+        Standard_Deviation _ _ _ -> [0, 0, 0]
+        Concatenate _ _ _ _ _ -> Nothing
+        First _ _ -> Nothing
+        Last _ _ -> Nothing
+        _ -> 0
+
+    make_aggregator : Table->(Any->Integer->Any)
+    make_aggregator table =
+        create_closure c function =
+            col = this.resolve_column table c
+            function col
+
+        is_empty s = if s.is_nothing then True else case s of
+            Text -> s.is_empty
+            _ -> Error.throw (Invalid_Aggregation_Method this.col "Empty is only valid for Text")
+
+        case this of
+            Count _ -> count->_->(count+1)
+            Count_Not_Null c _ -> create_closure c col->count->i->(count + if (col.at i).is_nothing then 0 else 1)
+            Count_Null c _ -> create_closure c col->count->i->(count + if (col.at i).is_nothing then 1 else 0)
+            Count_Not_Empty c _ -> create_closure c col->count->i->(count + if is_empty (col.at i) then 0 else 1)
+            Count_Empty c _ -> create_closure c col->count->i->(count + if is_empty (col.at i) then 1 else 0)
+            First c _ -> create_closure c col->current->i->(if i==0 then (col.at i) else current)
+            Last c _ -> create_closure c col->_->i->(col.at i)
+            Sum c _ -> create_closure c col->total->i->
+                v = col.at i
+                (total + if v.is_nothing then 0 else v)
+            Average c _ -> create_closure c col->a->i->
+                v = col.at i
+                (c + if v.is_nothing then a else [(a.at 0) + 1, (a.at 1) + v])
+            Standard_Deviation c _ _ -> create_closure c col->a->i->
+                v = col.at i
+                (c + if v.is_nothing then a else [(a.at 0) + 1, (a.at 1) + v, (a.at 2) + v*v])
+            Concatenate c _ j _ _ -> create_closure c col->text->i->
+                val=if (col.at i).is_nothing then "" else (col.at i).to_text
+                if i==0 then val else (text + j + val)
+            Median c _ -> create_closure c col->map->i->
+                val = col.at i
+                if val.is_nothing then map else (map.insert val (1 + (map.get_or_else val 0)))
+            Count_Distinct columns _ ->
+                resolved = columns.map c->(this.resolve_column table c)
+                key_maker i = Group_By_Key (resolved.map c->(c.at i))
+                map->i->(map.insert (key_maker i) 1)
+
+    evaluate : Any->Any
```

Comment:
I'm a bit skeptical about this split of `initial_value`, `aggregator` and `evaluate` - requires a lot of 'looking around' to understand the algorithm for a given function.

As this is a prototype and will be changed in the future - I do _not_ suggest to change anything now.

But in the future for similar use cases, I would suggest doing this in a more "local" way, for example we can define a helper type:
```
type Aggregation_Algorithm initializer aggregator evaluator
```

and then create the algorithms in one place:
```
algorithm : Aggregation_Algorithm
algorithm = case this of
    ...
    Last _ -> 
        initializer = ...
        aggregator = ...
        evaluator = ...
```

This way all parts of each algorithm are next to each other so it is much easier to grasp a given algorithm and see if it is correct. If the algorithm is spread throughout multiple places it is harder to spot potential issues.